internal/routes/attempt.go

```diff
@@ -0,0 +1,95 @@
+package routes
+
+import (
+	"github.com/MergeMinds/mm-backend-go/internal/attempt"
+	"github.com/go-fuego/fuego"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+)
+
+type AttemptHandler interface {
+	CreateAttempt(c fuego.ContextWithBody[attempt.MakeAttempt]) (*attempt.Attempt, error)
+	GetAttempt(c fuego.ContextNoBody) (*attempt.AttemptResponse, error)
+	GradeAttempt(c fuego.ContextWithBody[attempt.Attempt]) (*struct{}, error)
+	// UpdatedAttempt(c fuego.ContextWithBody[attempt.AttemptUpdate]) (*attempt.Attempt, error)
+	DeleteAttempt(c fuego.ContextNoBody) (any, error)
+}
+
+type attemptHandler struct {
+	repo   attempt.AttemptRepository
+	logger *zap.Logger
+}
+
+func NewAttemptHandler(repo attempt.AttemptRepository, logger *zap.Logger) AttemptHandler {
+	return &attemptHandler{
+		repo: repo,
+	}
+}
+
+func (h *attemptHandler) CreateAttempt(c fuego.ContextWithBody[attempt.MakeAttempt]) (*attempt.Attempt, error) {
+	body, err := c.Body()
+	if err != nil {
+		return nil, fuego.BadRequestError{Title: "INVALID_JSON"}
```

Comment:
You use `fuego.BadRequestError` multiple times with same content, so right way is to create sentinel error `ErrInvalidJson` and send it as response.

Reply:
And the second thing here... Doesn't fuego handle that for us automatically?